Reorganise, reformat and package

johnomotani opened the pull request:
* Move source files into `hypnotoad2` subdirectory
    * organise into several subdirectories
* Format with `black` and `flake8`
* Provide a `setup.py` to make package installable

Lots of changes, but almost all formatting. The only code changes are (i) fixing some imports that were broken by files being moved and (ii) converting assignment of `lambda`s like `f = lambda s: s**2` into function definitions
```
def f(s):
    return s**2
```
because `flake8` complained about the `lambda`s. This should not change any functionality though.

Once we get CI set up, I'm planning to require `black .` and `flake8` to pass the code before merging any future PRs. `xarray` does this and I think it's 'a good thing :tm:'.

Closes #3.

ZedThree commented:
The version can be done in a similar way to `long_description` in `setup.py`. Have a file `hypnotoad2/__version__.py` that contains `__version__ = "0.1.0"`:

https://github.com/ZedThree/pyxpad/blob/master/pyxpad/_version.py

https://github.com/ZedThree/pyxpad/blob/bf31cb5ddae8c086677df2b75379a8af9dcb45eb/setup.py#L3-L7

This means there's only one place that needs updating, plus now the installed package has that metadata too.

johnomotani commented:
Thanks @ZedThree, that's a nice maintainable way to keep the version number. I've added that, and also updated the license information in setup.py so that `pip show hypnotoad2` displays a license instead of `License: UNKNOWN`. Could you re-approve please?
